tests/transforms/test_metric_tensor.py

```diff
@@ -618,119 +594,528 @@ def test_multi_qubit_gates(self):
+                0.93,
+                0.14,
+                0.46,
+                0.31,
+                0.83,
+                0.79,
+                0.25,
+                0.40,
+                0.16,
+            ],
+            (2, 3, 3),
+            requires_grad=True,
+        ),
+    ),
+    (-0.1111, -0.2222),
+    (-0.1111, -0.2222, 0.4554),
+    (-0.1735, -0.1735, -0.2846, -0.2846),
+    (-0.1735, -0.2846),
+    (-0.1735, -0.2846),
+    (-0.1735,),
+    (-0.1111, 0.3333),
+]
+
+
+def autodiff_metric_tensor(ansatz, num_wires):
+    """Compute the metric tensor by full state vector
+    differentiation via autograd."""
+    dev = qml.device("default.qubit", wires=num_wires)
+
+    @qml.qnode(dev)
+    def qnode(*params):
+        ansatz(*params, wires=dev.wires)
+        return qml.state()
+
+    def mt(*params):
+        state = qnode(*params)
+        rqnode = lambda *params: np.real(qnode(*params))
+        iqnode = lambda *params: np.imag(qnode(*params))
+        jac = qml.jacobian(rqnode)(*params) + 1j * qml.jacobian(iqnode)(*params)
+        psidpsi = np.tensordot(np.conj(state), jac, axes=([0], [0]))
+        return np.real(
+            np.tensordot(np.conj(jac), jac, axes=([0], [0]))
+            - np.tensordot(np.conj(psidpsi), psidpsi, axes=0)
+        )
+
+    return mt
+
+
+class TestFullMetricTensor:
+
+    num_wires = 3
+
+    @pytest.mark.parametrize("ansatz, params", zip(fubini_ansatze, fubini_params))
+    def test_correct_output_autograd(self, ansatz, params):
+        expected = autodiff_metric_tensor(ansatz, self.num_wires)(*params)
+        dev = qml.device("default.qubit.autograd", wires=self.num_wires + 1)
+
+        @qml.qnode(dev, interface="autograd")
+        def circuit(*params):
+            """Circuit with dummy output to create a QNode."""
+            ansatz(*params, dev.wires[:-1])
+            return qml.expval(qml.PauliZ(0))
+
+        mt = qml.metric_tensor(circuit, approx=None)(*params)
+
+        assert np.allclose(mt, expected)
+
+    @pytest.mark.skip(reason="JAX does not support the forward pass metric tensor.")
+    @pytest.mark.parametrize("ansatz, params", zip(fubini_ansatze, fubini_params))
+    def test_correct_output_jax(self, ansatz, params):
+        jax = pytest.importorskip("jax")
+        from jax import numpy as jnp
+
+        expected = autodiff_metric_tensor(ansatz, self.num_wires)(*params)
+        dev = qml.device("default.qubit.jax", wires=self.num_wires + 1)
+
+        params = tuple(jnp.array(p) for p in params)
+
+        @qml.qnode(dev, interface="jax")
+        def circuit(*params):
+            """Circuit with dummy output to create a QNode."""
+            ansatz(*params, dev.wires[:-1])
+            return qml.expval(qml.PauliZ(0))
+
+        mt = qml.metric_tensor(circuit, approx=None)(*params)
+
+        assert np.allclose(mt, expected)
+
+    @pytest.mark.skip(
+        reason="The torch implementation is not adapted to the forward pass metric tensor yet."
+    )

```

Comment:
Doesn't it work if you set `requires_grad=True`?

Reply:
Almost, I think. I set this on hold inbetween, so I will return to it now. But it should be possible to make this work without a huge hassle.